nano/core_test/node.cpp

```diff
@@ -3153,6 +3217,7 @@ TEST (confirmation_height, prioritize_frontiers)
 	transaction.refresh ();
 	node->active.prioritize_frontiers_for_confirmation (transaction, std::chrono::seconds (1), std::chrono::seconds (1));
 	ASSERT_TRUE (priority_orders_match (node->active.priority_wallet_cementable_frontiers, std::array<nano::account, num_accounts>{ key3.pub, nano::genesis_account, key4.pub, key1.pub, key2.pub }));
+	node->active.confirm_frontiers (transaction);
```

Comment:
I don't think you need the scoped block below anymore (`node->active.next_frontier_check = ...`). It was overriding the frontier check so that in the next request loop it would call `confirm_frontiers` (which you are now calling manually) which wouldn't do anything because the `frontier_confirmation` is set to `disabled`.